crates/ln-dlc-storage/src/lib.rs

```diff
@@ -0,0 +1,1191 @@
+            .get_channel(&channel_id)
+            .expect("could not retrieve previously inserted channel.");
+
+        storage
+            .delete_channel(&channel_id)
+            .expect("to be able to delete the channel");
+
+        assert!(storage
+            .get_channel(&channel_id)
+            .expect("error getting channel.")
+            .is_none());
+    }
+
+    #[test]
+    fn persist_chain_monitor_test() {
+        let storage = DlcStorageProvider::new(InMemoryDLCStoreProvider::new());
+        let chain_monitor = ChainMonitor::new(123);
+
+        storage
+            .persist_chain_monitor(&chain_monitor)
+            .expect("to be able to persist the chain monistor.");
+
+        let retrieved = storage
+            .get_chain_monitor()
+            .expect("to be able to retrieve the chain monitor.")
+            .expect("to have a persisted chain monitor.");
+
+        assert_eq!(chain_monitor, retrieved);
+    }
+
+    #[test]
+    fn get_offered_sub_channels_only_offered() {
+        let mut storage = DlcStorageProvider::new(InMemoryDLCStoreProvider::new());
+        insert_sub_channels(&mut storage);
+
+        let offered_sub_channels = storage
+            .get_offered_sub_channels()
+            .expect("Error retrieving offered sub channels");
+        assert_eq!(2, offered_sub_channels.len());
+    }
+
+    #[test]
+    fn get_sub_channels_all_returned() {
+        let mut storage = DlcStorageProvider::new(InMemoryDLCStoreProvider::new());
+        insert_sub_channels(&mut storage);
+
+        let offered_sub_channels = storage
+            .get_sub_channels()
+            .expect("Error retrieving offered sub channels");
+        assert_eq!(4, offered_sub_channels.len());
+    }
+
+    #[test]
+    fn save_actions_roundtip_test() {
+        let storage = DlcStorageProvider::new(InMemoryDLCStoreProvider::new());
+        let actions: Vec<_> =
+            serde_json::from_str(include_str!("../test_files/sub_channel_actions.json")).unwrap();
+        storage
+            .save_sub_channel_actions(&actions)
+            .expect("Error saving sub channel actions");
+        let recovered = storage
+            .get_sub_channel_actions()
+            .expect("Error getting sub channel actions");
+        assert_eq!(actions, recovered);
+    }
+
+    #[test]
+    fn get_actions_unset_test() {
+        let storage = DlcStorageProvider::new(InMemoryDLCStoreProvider::new());
+        let actions = storage
+            .get_sub_channel_actions()
+            .expect("Error getting sub channel actions");
+        assert_eq!(actions.len(), 0);
+    }
+
+    #[test]
+    fn get_empty_actions_test() {
+        let storage = DlcStorageProvider::new(InMemoryDLCStoreProvider::new());
+        storage.save_sub_channel_actions(&[]).unwrap();
+        let actions = storage
+            .get_sub_channel_actions()
+            .expect("Error getting sub channel actions");
+        assert_eq!(actions.len(), 0);
+    }
+}
```

Comment:
I guess this file was copied from somewhere?

Reply:
The majority of the code was copied from the `dlc-sled-storage-provider` and adapted to allow to abstract the storage layer.